qcfractal/tests/test_collections.py

```diff
@@ -132,6 +212,132 @@ def test_gradient_dataset_statistics(gradient_dataset_fixture):
+    ds.add_entry("He", ptl.Molecule.from_data("He -1.1 0 0"))
+    ds.units = "hartree"
+    ds.save()
+
+    energy = {
+        "name": "Fake Energy",
+        "theory_level": "pseudo-random values",
+        "values": {
+            "He1": 1234.5,
+            "He": 5.4321
+        },
+        "theory_level_details": {
+            "driver": "energy",
+            "program": "fake_program",
+            "basis": "fake_basis",
+            "method": "fake_method",
+        },
+        "units": "hartree"
+    }
+    gradient = {
+        "name": "Fake Gradient",
+        "theory_level": "pseudo-random values",
+        "values": {
+            "He1": [0.03, 0, 0.02, -0.02, 0, -0.03],
+            "He": [0.03, 0, 0.02]
+        },
+        "theory_level_details": {
+            "driver": "gradient",
+            "program": "fake_program",
+            "basis": "fake_basis",
+            "method": "fake_method",
+        },
+        "units": "hartree/bohr"
+    }
+    hessian = {
+        "name": "Fake Hessian",
+        "theory_level": "pseudo-random values",
+        "values": {
+            "He1": list(np.eye(6).ravel()),
+            "He": [1, 0.2, 0.1, 0.2, 1, 0.4, 0.1, 0.4, 1]
+        },
+        "theory_level_details": {
+            "driver": "hessian",
+            "program": "fake_program",
+            "basis": "fake_basis",
+            "method": "fake_method",
+        },
+        "units": "hartree/bohr**2"
+    }
+    dipole = {
+        "name": "Fake Dipole",
+        "theory_level": "pseudo-random values",
+        "values": {
+            "He": [1., 2., 3.],
+            "He1": [1., -2., 0.]
+        },
+        "theory_level_details": {
+            "driver": "dipole",
+            "program": "fake_program",
+            "basis": "fake_basis",
+            "method": "fake_method",
+        },
+        "units": "e * bohr"
+    }
+    energy_FF = {
+        "name": "Fake FF Energy",
+        "theory_level": "some force field",
+        "values": {
+            "He1": 0.5,
+            "He": 42.
+        },
+        "theory_level_details": {
+            "driver": "energy",
+            "program": "fake_program",
+            "basis": None,
+            "method": "fake_method"
+        },
+        "units": "hartree"
+    }
+
+    ds.add_contributed_values(energy)
+    ds.add_contributed_values(gradient)
+    ds.add_contributed_values(hessian)
+    ds.add_contributed_values(dipole)
+    ds.add_contributed_values(energy_FF)
+
+    with pytest.raises(KeyError):
+        ds.add_contributed_values(energy)
+
+    ds.save()
+
+    yield client, client.get_collection("dataset", "ds_contributed")
+
+
+def test_dataset_contributed_units(contributed_dataset_fixture):
+    _, ds = contributed_dataset_fixture
+
+    assert ds._column_metadata[ds.get_values(name="Fake Energy").columns[0]]["units"] == "kcal / mol"
+    assert ds._column_metadata[ds.get_values(name="Fake Gradient").columns[0]]["units"] == "hartree/bohr"
```

Comment:
Careful here since you have `hartree/bohr` and `kcal / mol`, granted doesn't matter or our units system. I wonder if these things should live as `pint` units here rather than strings. A problem for another time, but worth thinking about.